app/src/test/java/com/keylesspalace/tusky/util/HttpHeaderLinkTest.kt

```diff
@@ -0,0 +1,83 @@
+package com.keylesspalace.tusky.util
+
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import org.junit.Assert.assertEquals
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.robolectric.annotation.Config
+import kotlin.reflect.full.primaryConstructor
+import kotlin.reflect.jvm.isAccessible
+
+@Config(sdk = [28])
+@RunWith(AndroidJUnit4::class)
+class HttpHeaderLinkTest {
+    data class TestData(val name: String, val input: String, val want: List<HttpHeaderLink>)
+
+    @Test
+    fun shouldParseValidLinks() {
+        // Constructor is private, so call through reflection
+        val constructor = HttpHeaderLink::class.primaryConstructor!!
```

Comment:
Usually we just make public what we need and annotate it with `@VisibleForTesting`